### PR TITLE
Features C/JS API

### PR DESCRIPTION
**Pull request by tlively**
Add feature handling to the C/JS APIs. No features are enabled by
default, so all used features will have to be explicitly enabled in
order for modules to validate.

### CODE REVIEW

File: src/js/binaryen.js-post.js [hunk 1]
```diff
@@ -84,6 +84,14 @@ Module['ExternalTable'] = Module['_BinaryenExternalTable']();
 Module['ExternalMemory'] = Module['_BinaryenExternalMemory']();
 Module['ExternalGlobal'] = Module['_BinaryenExternalGlobal']();
 
+// Features
+Module['FeatureAtomics'] = Module['_BinaryenFeatureAtomics']();
+Module['FeatureBulkMemory'] = Module['_BinaryenFeatureBulkMemory']();
+Module['FeatureMutableGlobals'] = Module['_BinaryenFeatureMutableGlobals']();
+Module['FeatureNontrappingFPToInt'] = Module['_BinaryenFeatureNontrappingFPToInt']();
+Module['FeatureSignExt'] = Module['_BinaryenFeatureSignExt']();
+Module['FeatureSIMD128'] = Module['_BinaryenFeatureSIMD128']();
```
Comment (tlively) [inline src/js/binaryen.js-post.js]: Are these names idiomatic enough?

Comment (kripken) [reply src/js/binaryen.js-post.js]: How about an API that looks like `Binaryen.Features.Atomics`?

cc @dcodeIO

Comment (dcodeIO) [reply src/js/binaryen.js-post.js]: Sounds good :)

**Comment by tlively** [comment]
@kripken is that CI error meaningful?

**Comment by kripken** [comment]
The error should be fixed now on emscripten - will be fixed in CI soon too. Given this PR is exactly about the JS API, we should probably wait on that.

**Comment by tlively** [comment]
@kripken Any more comments on this?

**Pull request merged**